timemachine/fe/rest/single_topology.py

```diff
@@ -0,0 +1,143 @@
+from dataclasses import replace
+from functools import cached_property
+from typing import Callable, Literal
+
+import jax.numpy as jnp
+import numpy as np
+from numpy.typing import NDArray
+from openmm import app
+
+from timemachine.fe.single_topology import SingleTopology
+from timemachine.fe.system import HostGuestSystem, VacuumSystem
+from timemachine.ff import Forcefield
+from timemachine.potentials import HarmonicAngleStable, NonbondedPairListPrecomputed
+
+from .bond import CanonicalBond, mkbond
+from .interpolation import Exponential, InterpolationFxn, Linear, Quadratic, Symmetric
+from .queries import get_rotatable_bonds
+
+InterpolationFxnName = Literal["linear", "quadratic", "exponential"]
+
+
+def get_temperature_scale_interpolation_fxn(
+    max_temperature_scale: float, interpolation_fxn: InterpolationFxnName
+) -> InterpolationFxn:
+    make_interp_fxn: Callable[[float, float], InterpolationFxn]
+    match interpolation_fxn:
+        case "linear":
+            make_interp_fxn = Linear
+        case "quadratic":
+            make_interp_fxn = Quadratic
+        case "exponential":
+            make_interp_fxn = Exponential
+
+    return Symmetric(make_interp_fxn(1.0, max_temperature_scale))
+
+
+class SingleTopologyREST(SingleTopology):
+    def __init__(
+        self,
+        mol_a,
+        mol_b,
+        core,
+        forcefield,
+        max_temperature_scale: float,
+        temperature_scale_interpolation_fxn: InterpolationFxnName = "exponential",
+    ):
+        super().__init__(mol_a, mol_b, core, forcefield)
```

Comment:
Add docs for max_temperature_scale, temperature_scale_interpolation_fxn

Reply:
Documented in https://github.com/proteneer/timemachine/pull/1424/commits/54953cdcb6c8a62077ef1a0853190715703eee31